stream.go

```diff
@@ -168,8 +168,24 @@ func NewClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 }
 
 func newClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, method string, opts ...CallOption) (_ ClientStream, err error) {
-	if md, _, ok := metadata.FromOutgoingContextRaw(ctx); ok {
-		if err := imetadata.Validate(md); err != nil {
+	// validate original input
+	if md, added, ok := metadata.FromOutgoingContextRaw(ctx); ok {
+		// copy the original MD to avoid changing it
+		mdSize := len(md)
+		for i := range added {
+			mdSize += len(added[i]) / 2
+		}
+		merged := make(metadata.MD, mdSize)
+		for k, v := range md {
+			merged[k] = v
+		}
+		for _, pairs := range added {
+			for i := 0; i < len(pairs); i += 2 {
+				k := pairs[i]
+				merged[k] = append(merged[k], pairs[i+1])
+			}
+		}
+		if err := imetadata.Validate(merged); err != nil {
```

Comment:
Instead of this, would you mind adding a `metadata.ValidateRaw` function, or equivalent, that can validate `md` and `added` (or just `added` alone) without needing to allocate memory and make a copy?

Reply:
> Instead of this, would you mind adding a `metadata.ValidateRaw` function, or equivalent, that can validate `md` and `added` (or just `added` alone) without needing to allocate memory and make a copy?

Very sound advice, will do it